pilot/pkg/kube/inject/inject.go

```diff
@@ -46,65 +48,50 @@ import (
-		{"readiness.status.sidecar.istio.io/failureThreshold", validateUInt32},
-		{"readiness.status.sidecar.istio.io/applicationPorts", validateReadinessApplicationPorts},
-		{"traffic.sidecar.istio.io/includeOutboundIPRanges", ValidateIncludeIPRanges},
-		{"traffic.sidecar.istio.io/excludeOutboundIPRanges", ValidateExcludeIPRanges},
-		{"traffic.sidecar.istio.io/includeInboundPorts", ValidateIncludeInboundPorts},
-		{"traffic.sidecar.istio.io/excludeInboundPorts", ValidateExcludeInboundPorts},
-		{"traffic.sidecar.istio.io/kubevirtInterfaces", alwaysValidFunc},
-	}
-
-	annotationPolicy = annotationRegistry[0]
-	annotationStatus = annotationRegistry[1]
+	annotationRegistry = map[string]annotationValidationFunc{
+		annotations.RegisterAnnotation(annotationPolicy, "").Name:                                        alwaysValidFunc,
+		annotations.RegisterAnnotation(annotationStatus, "").Name:                                        alwaysValidFunc,
+		annotations.RegisterAnnotation("sidecar.istio.io/proxyImage", "").Name:                           alwaysValidFunc,
+		annotations.RegisterAnnotation("sidecar.istio.io/interceptionMode", "").Name:                     validateInterceptionMode,
+		annotations.RegisterAnnotation("status.sidecar.istio.io/port", "").Name:                          validateStatusPort,
+		annotations.RegisterAnnotation("readiness.status.sidecar.istio.io/initialDelaySeconds", "").Name: validateUInt32,
+		annotations.RegisterAnnotation("readiness.status.sidecar.istio.io/periodSeconds", "").Name:       validateUInt32,
+		annotations.RegisterAnnotation("readiness.status.sidecar.istio.io/failureThreshold", "").Name:    validateUInt32,
+		annotations.RegisterAnnotation("readiness.status.sidecar.istio.io/applicationPorts", "").Name:    validateReadinessApplicationPorts,
+		annotations.RegisterAnnotation("traffic.sidecar.istio.io/includeOutboundIPRanges", "").Name:      ValidateIncludeIPRanges,
+		annotations.RegisterAnnotation("traffic.sidecar.istio.io/excludeOutboundIPRanges", "").Name:      ValidateExcludeIPRanges,
+		annotations.RegisterAnnotation("traffic.sidecar.istio.io/includeInboundPorts", "").Name:          ValidateIncludeInboundPorts,
+		annotations.RegisterAnnotation("traffic.sidecar.istio.io/excludeInboundPorts", "").Name:          ValidateExcludeInboundPorts,
+		annotations.RegisterAnnotation("traffic.sidecar.istio.io/kubevirtInterfaces", "").Name:           alwaysValidFunc,
+	}
 )
 
-type annotationValidationFunc func(value string) error
-
 func validateAnnotations(annotations map[string]string) (err error) {
-	for _, validator := range annotationRegistry {
-		if e := validator.validate(annotations); e != nil {
-			err = multierror.Append(err, e)
+	for name, value := range annotations {
+		if v, ok := annotationRegistry[name]; ok {
+			if e := v(value); e != nil {
+				err = multierror.Append(err, fmt.Errorf("invalid value '%s' for annotation '%s': %v", value, name, e))
+			}
+		} else {
+			// unknown annotation, ignore
```

Comment:
should we log?

Reply:
Seems like the original logic skips unknown annotations.

Reply:
I don't think we want to log this. It's totally legit to have non-Istio annotations there and we don't want to spam the log with thousands of useless warnings.

So how about we log if the annotation name includes "istio" in it? This would indicate likely a misspelled annotation name.

Reply:
Done.

Reply:
good point ... I'd be fine doing nothing, but logging if it contains "istio" is fine too.